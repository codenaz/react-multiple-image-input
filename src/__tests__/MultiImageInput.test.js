import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent } from '@testing-library/react';
import MultiImageInput from '../lib';

const crop = {
  unit: '%',
  width: '100'
};
let images = {};
const setImages = imageUpdate => (image = imageUpdate);

test('renders correctly', () => {
  let images = {};
  const setImages = imageUpdate => (image = imageUpdate);

  const { baseElement } = render(
    <MultiImageInput
      images={images}
      setImages={setImages}
      cropConfig={{ crop, ruleOfThirds: true, aspect: 4 / 3 }}
    />
  );
  expect(baseElement).toMatchSnapshot();
});

test('renders the right number of images', () => {
  let images = {
    0: 'https://dababy.com/image.png',
    1: 'https://larajean.com/image.png'
  };
  const setImages = imageUpdate => (image = imageUpdate);
  const { getAllByAltText } = render(
    <MultiImageInput
      images={images}
      setImages={setImages}
      cropConfig={{ crop, ruleOfThirds: true, aspect: 4 / 3 }}
    />
  );
  const selectedImages = getAllByAltText(/^uploaded image\d/);
  expect(selectedImages.length).toBe(2);
});

test('renders the uploadImage element', () => {
  let images = {
    0: 'https://dababy.com/image.png',
    1: 'https://larajean.com/image.png'
  };
  const setImages = imageUpdate => (image = imageUpdate);
  const { getByText } = render(
    <MultiImageInput
      images={images}
      setImages={setImages}
      cropConfig={{ crop, ruleOfThirds: true, aspect: 4 / 3 }}
    />
  );
  const uploadImage = getByText('ADD IMAGE');
  expect(uploadImage).not.toBeNull();
});

test('deletes an image', () => {
  let images = {
    0: 'https://dababy.com/image.png',
    1: 'https://larajean.com/image.png',
    2: 'hhh'
  };
  const setImages = jest.fn(imageUpdate => (images = imageUpdate));
  const { getAllByRole } = render(
    <MultiImageInput
      images={images}
      setImages={setImages}
      cropConfig={{ crop, ruleOfThirds: true, aspect: 4 / 3 }}
    />
  );
  const deleteButton = getAllByRole('button', { label: 'Delete Image 0' });
  fireEvent.click(deleteButton[1]);
  expect(setImages).toBeCalled();
  expect(Object.keys(images).length).toBe(2);
});

test('crops an image', () => {
  let images = {
    0: 'https://dababy.com/image.png',
    1: 'https://larajean.com/image.png',
    2: 'hhh'
  };
  const setImages = jest.fn(imageUpdate => (images = imageUpdate));
  const { getAllByRole, getByText } = render(
    <MultiImageInput
      images={images}
      setImages={setImages}
      cropConfig={{ crop, ruleOfThirds: true, aspect: 4 / 3 }}
    />
  );
  const editButton = getAllByRole('button', { label: 'Edit Image 0' });
  fireEvent.click(editButton[0]);
  const cropButton = getByText('Crop');
  expect(cropButton).not.toBeNull();
});
