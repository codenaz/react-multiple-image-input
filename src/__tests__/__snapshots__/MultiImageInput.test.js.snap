// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<body>
  <div>
    <div
      class="sc-gtLWhw sc-egkSDF dBuNdU fhGJvS"
    >
      <div
        class="sc-gtLWhw sc-fAUdSK dBuNdU hzsuBv"
      >
        <div
          role="button"
          style="cursor: pointer;"
        >
          <svg
            fill="none"
            height="46"
            style="margin-bottom: 0.5rem;"
            viewBox="0 0 58 46"
            width="58"
          >
            <path
              d="M57 0H1a1 1 0 00-1 1v44a1 1 0 001 1h56a1 1 0 001-1V1a1 1 0 00-1-1zm-1 44H2V2h54v42z"
              fill="rgba(255,255,255,0.6)"
              fill-opacity="0.6"
            />
            <path
              d="M16 22.138a5.575 5.575 0 005.57-5.568A5.575 5.575 0 0016 11a5.575 5.575 0 00-5.569 5.569 5.575 5.575 0 005.57 5.569zM16 13a3.574 3.574 0 013.57 3.569A3.574 3.574 0 0116 20.138a3.573 3.573 0 01-3.569-3.568 3.574 3.574 0 013.57-3.57zM7 40c.234 0 .47-.082.66-.25L23.973 25.39l10.302 10.3a.999.999 0 101.414-1.413l-4.807-4.807 9.181-10.054 11.261 10.323a1 1 0 001.351-1.475l-12-11a1.031 1.031 0 00-.72-.262 1.002 1.002 0 00-.694.325l-9.794 10.727-4.743-4.743a1 1 0 00-1.368-.044L6.339 38.249A1 1 0 007 39.999z"
              fill="rgba(255,255,255,0.6)"
              fill-opacity="0.6"
            />
          </svg>
          <span
            class="sc-kLhKbu eYeunN"
            color="outlineColor"
            font-size="small"
            font-weight="normal"
            style="display: block;"
          >
            ADD IMAGE
          </span>
        </div>
        <input
          accept="image/*"
          multiple=""
          style="visibility: hidden;"
          type="file"
        />
      </div>
    </div>
  </div>
</body>
`;
